packages/@aws-cdk/aws-kinesisfirehose-destinations/test/s3-bucket.test.ts

```diff
@@ -0,0 +1,226 @@
+import '@aws-cdk/assert-internal/jest';
+import { ABSENT, Capture, MatchStyle, ResourcePart, anything, arrayWith } from '@aws-cdk/assert-internal';
+import * as iam from '@aws-cdk/aws-iam';
+import * as firehose from '@aws-cdk/aws-kinesisfirehose';
+import * as logs from '@aws-cdk/aws-logs';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import * as firehosedestinations from '../lib';
+
+describe('S3 destination', () => {
+  let stack: cdk.Stack;
+  let bucket: s3.IBucket;
+  let destinationRole: iam.IRole;
+
+  beforeEach(() => {
+    stack = new cdk.Stack();
+    bucket = new s3.Bucket(stack, 'Bucket');
+    destinationRole = new iam.Role(stack, 'Destination Role', {
+      assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+    });
+  });
+
+  it('provides defaults when no configuration is provided', () => {
+    new firehose.DeliveryStream(stack, 'DeliveryStream', {
+      destinations: [new firehosedestinations.S3Bucket(bucket, { role: destinationRole })],
+    });
+
+    expect(stack).toHaveResource('AWS::KinesisFirehose::DeliveryStream', {
+      ExtendedS3DestinationConfiguration: {
+        BucketARN: stack.resolve(bucket.bucketArn),
+        CloudWatchLoggingOptions: {
+          Enabled: true,
+          LogGroupName: anything(),
+          LogStreamName: anything(),
+        },
+        RoleARN: stack.resolve(destinationRole.roleArn),
+      },
+    });
+    expect(stack).toHaveResource('AWS::Logs::LogGroup');
+    expect(stack).toHaveResource('AWS::Logs::LogStream');
+  });
+
+  it('creates a role when none is provided', () => {
+    const capturedRoleArn = Capture.aString();
```

Comment:
I would actually put the logical ID of the Role directly in the test. I understand it will be an ugly, hashed logical ID. But I think that's good: in general, you want the tests to act as a guard to make sure the logical IDs of elements in the template you generate are stable while you're doing refactorings (it's easy to change them when changing the construct tree when refactoring).

I understand changing this ID in theory would not be a breaking change, because Roles are statelss, but in my opinion you should be forced to update the unit tests when that happens (in other words, this should be a conscious decision, not something that happens accidentally).

Same comment applies to the tests below.